contracts/lib/CustomSafeERC20.sol

```diff
@@ -0,0 +1,55 @@
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+/**
+ * @title SafeERC20
+ * @dev Wrappers around ERC20 operations that throw on failure.
+ * To use this library you can add a `using SafeERC20 for ERC20;` statement to your contract,
+ * which allows you to call the safe operations as `token.safeTransfer(...)`, etc.
+ */
+library SafeERC20 {
+  function safeTransfer(
+    IERC20 token,
+    address to,
+    uint256 value
+  )
+    internal
+  {
+    require(token.transfer(to, value));
+  }
+
+  function safeTransferFrom(
+    IERC20 token,
+    address from,
+    address to,
+    uint256 value
+  )
+    internal
+  {
+    require(token.transferFrom(from, to, value));
+  }
+
+  function safeApprove(IERC20 token, address spender, uint256 value, bytes memory errPrefix) internal {
+    if (_callOptionalReturn(token, abi.encodeWithSelector(token.approve.selector, spender, value))) {
+        return;
+    }
+    require(_callOptionalReturn(token, abi.encodeWithSelector(token.approve.selector, spender, 0))
+        && _callOptionalReturn(token, abi.encodeWithSelector(token.approve.selector, spender, value)),
+        string(abi.encodePacked(errPrefix, 'ERC20 approve failed')));
```

Comment:
What if we skipped the use of `errPrefix` entirely? We don't even use it ourselves as the library consumer. This should save the cost of doing the `abi.encodePacked`